src/core/mount.c

```diff
@@ -1603,6 +1604,13 @@ static int mount_setup_existing_unit(
         MountProcFlags flags =
                 MOUNT(u)->proc_flags | MOUNT_PROC_IS_MOUNTED;
 
+        /* If the proc mountinfo device is NULL or different from the passed in device
+         * the device needs to be updated.
+         */
+        if (!MOUNT(u)->parameters_proc_self_mountinfo.what ||
+            !streq(MOUNT(u)->parameters_proc_self_mountinfo.what, what))
```

Comment:
use streq_ptr(), and you can drop the first part f the check

Reply:
hmm, maybe it would be prettier to modify update_parameters_proc_self_mountinfo() to return as return value the flag combination to use here, i.e. either 0, or MOUNT_PROC_JUST_CHANGED or both MOUNT_PROC_JUST_CHANGED|MOUNT_PROC_DEVICE_CHANGED.

After all, what you are doing here is very similar to what update_parameters_proc_self_mountinfo() does, just one line further down, except that you just check one of the three fields, instead of all three

Reply:
Yes, that's a good suggestion.